be2-scala/src/main/scala/ch/epfl/pop/config/RuntimeEnvironment.scala

```diff
@@ -33,15 +33,47 @@ object RuntimeEnvironment {
 
     val virtualMachineParam = "scala.config"
     val pathConfig = sp(virtualMachineParam)
-    if (pathConfig != null && !pathConfig.trim.isEmpty) {
+    if (pathConfig != null && pathConfig.trim.nonEmpty) {
       pathConfig.trim
     } else {
       throw new RuntimeException(s"-D$virtualMachineParam was not provided.")
     }
   }
 
-  private lazy val appConfFile = getConfDir + File.separator + "application.conf"
+  private def testMode: Boolean = {
+    sp("test") != null
+  }
+
+  private lazy val confDir: String = getConfDir
+  private lazy val appConfFile = confDir + File.separator + "application.conf"
+
+  lazy val isTestMode: Boolean = testMode
+  lazy val serverPeersListPath: String =
+    if (isTestMode) {
+      confDir + File.separator + "server-peers-list-TEST.conf"
+    } else {
+      confDir + File.separator + "server-peers-list.conf"
+    }
 
   lazy val appConf: Config = ConfigFactory.parseFile(new File(appConfFile))
 
+  // Regex from dataGreetLao.json
+  private val addressPattern = "^(ws|wss)://.*(:d{0,5})?/.*$"
```

Comment:
The greetLao regex was changed in #1554, maybe we should put it in public in `RuntimEnvironment` so that it is  consistent across the project and easy to update ?

Reply:
Update: I say for now we leave this regex alone, I think a pr need to be made to fix greet_lao and greet_server regex as currently every address that do not contain `/` after the optional port are rejected....

Reply:
Ok let's fix this at once with the final regex version!